File: taskflow-fullstack/frontend/src/services/auth-service.ts
import { apiRequest } from '../lib/api-client'
import { clearSessionTokens, getRefreshToken, setSessionTokens } from '../lib/session'
import type {
  AuthResponse,
  CurrentUserResponse,
  LoginRequest,
  RegisterRequest,
} from '../types/auth'

export const authService = {
  async register(payload: RegisterRequest): Promise<AuthResponse> {
    const data = await apiRequest<AuthResponse>('/api/v1/auth/register', {
      method: 'POST',
      body: JSON.stringify(payload),
    })
    setSessionTokens(data.accessToken, data.refreshToken)
    return data
  },

  async login(payload: LoginRequest): Promise<AuthResponse> {
    const data = await apiRequest<AuthResponse>('/api/v1/auth/login', {
      method: 'POST',
      body: JSON.stringify(payload),
    })
    setSessionTokens(data.accessToken, data.refreshToken)
    return data
  },

  async logout(): Promise<void> {
    const refreshToken = getRefreshToken()

    if (!refreshToken) {
      clearSessionTokens()
      return
    }

    await apiRequest<void>('/api/v1/auth/logout', {
      method: 'POST',
      authenticated: true,
      body: JSON.stringify({ refreshToken }),
    })

    clearSessionTokens()
  },

  async getCurrentUser(): Promise<CurrentUserResponse> {
    return apiRequest<CurrentUserResponse>('/api/v1/auth/me', {
      method: 'GET',
      authenticated: true,
    })
  },

  async getHealth(): Promise<string> {
    return apiRequest<string>('/api/v1/auth/health', {
      method: 'GET',
    })
  },
}

File: taskflow-fullstack/backend/src/main/java/com/ruhatkaratas/taskflow/auth/service/AuthService.java
package com.ruhatkaratas.taskflow.auth.service;

import com.ruhatkaratas.taskflow.auth.dto.AuthResponse;
import com.ruhatkaratas.taskflow.auth.dto.CurrentUserResponse;
import com.ruhatkaratas.taskflow.auth.dto.LoginRequest;
import com.ruhatkaratas.taskflow.auth.dto.LogoutRequest;
import com.ruhatkaratas.taskflow.auth.dto.RegisterRequest;

public interface AuthService {

    AuthResponse register(RegisterRequest request);

    AuthResponse login(LoginRequest request);

    void logout(LogoutRequest request);

    CurrentUserResponse getCurrentUser(String email);
}
